en/index.md

```diff
@@ -524,6 +524,46 @@ For example:
 
 - **Don’t use:** Click the ![pencil](https://assets.ubuntu.com/v1/633aab2b-pencil.png) pencil icon to edit the post.
 
+## Hyperlinks
+
+Here are some pointers about the general usage of hyperlinks and how to format them correctly.
+
+### General usage
+Avoid excessive links in the same paragraph or instruction. If you find yourself introducing several links in your content, consider listing them in a separate section called "Related topics", "Additional resources", or similar.
+
+When possible, avoid external links. They can easily become outdated, making the document harder to maintain.
+
+When linking to versioned files or specific lines of code, copy the **permalink** instead of the URL. This will ensure the link is bound to the current revision of the file, so it will direct to the same content even if the file changes.
+
+### Formatting
+Make the link text match the exact title or heading that you are referencing. Avoid phrases like "this document", "this article", "click here".
+
+If linking to a section called "Formatting":
+
+- Use: `[Formatting](#formatting)` 
+- Don't use: `[Guidelines to format hyperlinks](#formatting)`
```

Comment:
I have some concerns about this particular guideline.
For reference, how-to guides, it is very possible we use similar structure across the different pages.
For example: https://documentation.ubuntu.com/lxd/en/latest/reference/image_format/#content

This may become quite generic and less helpful to readers.

I would suggest "avoid" instead of "don't use" as a recommendation instead.

Reply:
I also don't agree with this guideline for accessibility reasons. I never just use the header, because sometimes (if taken out of context) it makes it more confusing for the reader. I try to instead use my judgement to create link text that tells the user what the link contains to help them decide if they want to click it. 

i.e., the one you said to "don't use" is exactly what I use, because "formatting" as a single word is both too short to be a useful header for people with magnified screens, and because it's not descriptive enough to provide context. Formatting what exactly?

Reply:
Thanks a lot for the feedback! I think maybe these rules are a bit too concise and I'm making too many assumptions.

I'd never suggest using the plain header _unless it were entirely clear from the rest of the sentence's context_ (e.g. `See more information about formatting hyperlinks in [Formatting](#formatting)`), but that is not necessarily implied in the way I wrote that rule.

Here's my proposal:

```suggestion
Make the link text match the exact title or heading that you are referencing. Avoid phrases like "this document", "this article", "click here".

If linking to a section called "Formatting":

- Use: `[Formatting](#formatting)` 
- Don't use: `[Guidelines to format hyperlinks](#formatting)`
- Don't use: `[this section](#formatting)`
```
I do believe that the original header names should be kept whenever possible (i.e. whenever there's sufficient context in the sentence) -

Reply:
Thanks a lot for the feedback! I think maybe these rules are too concise or I'm making too many assumptions.

I'd never suggest using the header verbatim _unless it were entirely clear from the rest of the sentence's context_ (e.g. `See more information about formatting hyperlinks in [Formatting](#formatting)`), but I can see that it's not really implied in the rule I wrote.

I can also see some exceptions, so instead of going too deep into it, I propose softening the rule. Something like:

```markdown
When possible, try to make link text match the title or heading that you are referencing. If the title is not very descriptive on its own, the sentence should supplement additional context. 

Avoid phrases like "this document", "this article", "click here".

For example, when linking to a section called "Formatting":

- Use: `See the guidelines to format hyperlinks in the section [Formatting](#formatting).`
- Avoid: `See [Formatting](#formatting).` (no context)
- Avoid: `See the guidelines to format hyperlinks in [this section](#formatting).`
```

Reply:
I think this is fine for the example, but I agree that you often don't use the *exact* title, but rather a "combined" title (since the section title will often not have the context that is given by the page title).